compute/client_library/snippets/tests/test_ip_address.py

```diff
@@ -268,3 +269,26 @@ def test_release_static_ip(static_ip: Address):
     )
     ips = list_ip_addresses(client, PROJECT, region=region)
     assert static_ip.name not in ips
+
+
+def test_promote_ephemeral_ip(instance_with_ips: Instance):
+    ephemeral_ip = next(
+        (
+            config.nat_i_p
+            for interface in instance_with_ips.network_interfaces
+            for config in interface.access_configs
+            if config.type_ == "ONE_TO_ONE_NAT"
+        ),
+        None,
+    )
+
+    promote_ephemeral_ip(PROJECT, ephemeral_ip, REGION)
+
+    client = AddressesClient()
+    addresses_iterator = client.list(project=PROJECT, region=REGION)
+
+    for address in addresses_iterator:
+        # ex ephemeral ip in list of static IPs and still attached to instance
+        if address.address == ephemeral_ip and address.status == "IN_USE":
+            return
+    assert False, f"IP address {ephemeral_ip} was not promoted correctly"
```

Comment:
Since we're promoting this IP to a static one, it will live on, after the instance is deleted.

Make sure to delete the static IP at the end of this test using try-finally blocks.

Reply:
Or, if it's possible, maybe revert it back to ephemeral so instance deletion will release it.